src/faldbt/project.py

```diff
@@ -169,30 +175,46 @@ def get_models(self) -> List[DbtModel]:
     def get_tests(self) -> List[DbtTest]:
         return list(
             map(
-                lambda node: DbtTest(node),
+                DbtTest,
                 filter(
                     lambda node: node.resource_type == NodeType.Test,
                     self.nativeManifest.nodes.values(),
                 ),
             )
         )
 
-    def get_sources(self) -> List[ParsedSourceDefinition]:
-        return list(
-            map(lambda node: DbtSource(node=node), self.nativeManifest.sources.values())
-        )
+    def get_sources(self) -> List[DbtSource]:
+        return list(map(DbtSource, self.nativeManifest.sources.values()))
 
 
 @dataclass(init=False)
 class DbtRunResult:
     nativeRunResult: Optional[RunResultsArtifact]
-    results: Sequence[RunResultOutput]
 
     def __init__(self, nativeRunResult: Optional[RunResultsArtifact]):
-        self.results = []
         self.nativeRunResult = nativeRunResult
+
+    @property
+    def results(self) -> Sequence[RunResultOutput]:
         if self.nativeRunResult:
-            self.results = nativeRunResult.results
+            return self.nativeRunResult.results
+        else:
+            return []
+
+
+@dataclass(init=False)
```

Comment:
Not sure if this needs to be a dataclass

Reply:
Maybe we can remove the `__init__` and remove the `init=False` from the `@dataclass`? The `__init__` doesn't seem to add any value as is.